tests/test_deploy/TestControllers.py

```diff
@@ -1,32 +1,228 @@
+        sig = generate_signature(str(json.dumps({})).encode('utf-8'), g.github['ci_key'])
+        headers = generate_git_api_header('push', sig)
+        headers.remove('User-Agent')
+
+        response = self.app.test_client().post('/deploy', headers=headers, environ_overrides=WSGI_ENVIRONMENT)
+
+        self.assertEqual(response.status_code, 418)
+
+    def test_headers_invalid_User_Agent(self, mock_request_get):
         """
-        Test The View by sending a ping request
+        Test invalid user-agent beginning.
         """
-        response = self.app.test_client().post('/deploy', headers={'X-GitHub-Event': 'ping'})
+        mock_request_get.return_value.json.return_value = {"hooks": ['0.0.0.0']}
+        sig = generate_signature(str(json.dumps({})).encode('utf-8'), g.github['ci_key'])
+        headers = generate_git_api_header('push', sig)
+        headers['User-Agent'] = "invalid"
+
+        response = self.app.test_client().post('/deploy', headers=headers, environ_overrides=WSGI_ENVIRONMENT)
 
         self.assertEqual(response.status_code, 418)
-        self.assertIn("I\\\'m a teapot", str(response.data))
 
-    def test_headers_invalid_event(self):
+    def test_headers_event_not_push(self, mock_request_get):
         """
-        Test The View by sending an invalid event
+        Test The View by sending an event other than push.
         """
-        response = self.app.test_client().post('/deploy', headers={'X-GitHub-Event': 'Banana'})
+        mock_request_get.return_value.json.return_value = {"hooks": ['0.0.0.0']}
+        sig = generate_signature(str(json.dumps({})).encode('utf-8'), g.github['ci_key'])
+        headers = generate_git_api_header('pull', sig)
+
+        response = self.app.test_client().post('/deploy', headers=headers, environ_overrides=WSGI_ENVIRONMENT)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIn("Wrong event", str(response.data))
+
+    @mock.patch('mod_deploy.controllers.is_valid_signature', return_value=False)
+    @mock.patch('mod_deploy.controllers.g')
+    def test_headers_invalid_signature_event(self, mock_g, mock_valid_sign, mock_request_get):
+        """
+        Test the view by sending a valid event.
+        """
+        mock_request_get.return_value.json.return_value = {"hooks": ['0.0.0.0']}
+        self.app.config['INSTALL_FOLDER'] = gettempdir()
+        data = {
+            'ref': 'refs/heads/master'
+        }
+        sig = generate_signature(str(json.dumps(data)).encode('utf-8'), g.github['ci_key'])
+        headers = generate_git_api_header('push', sig)
+
+        with self.app.test_client() as client:
+            response = client.post(
+                '/deploy', environ_overrides=WSGI_ENVIRONMENT,
+                data=json.dumps(data), headers=headers
+            )
 
         self.assertEqual(response.status_code, 418)
-        self.assertIn("I\\\'m a teapot", str(response.data))
```

Comment:
This should stay in :joy:

Reply:
Oh, that's the default response we get :laughing: 

I'll let it stay then.

Reply:
I don't think we should let it stay since it's a redundant test. The more prominent reason why I don't want it to be there is it misleads the developer (it did me) to believe that it is a response from "our" server and not the default behavior. Hence, I believe only `status_code` should be there.